### DISCUSSION TITLE
React-Three-Fiber

### DISCUSSION THREAD

**Discussion by MaximilianFranz** [bldrs-ai/Share#371] [Q&A] [answered]
Hey all, 

cool tool you're building here! Thanks for that important work. 
I saw you're using react and three-js and wondered if you considered using react-three-fiber to integrate the scene state into the react lifecycle? 

This is an interesting discussion when it comes to scaling scene state with complex interactions with application state. 

What's the overall idea for state-management when interacting with the viewer? Especially when extending the plattform? 

Thanks already,

**Comment by OlegMoshkovich** [comment]
@pablo-mayrgundter please reply.

**Comment by pablo-mayrgundter** [comment, marked as answer]
Hi Maximillian!

Thanks for your interest.  We'd love to use r3f.  Main impediment currently is that we're accessing three.js indirectly, via ifc.js.  We had punted on r3f until we heard more on any plans by ifc.js.  Looking now I see some movement here:

https://github.com/IFCjs/web-ifc-viewer/issues/144

Are you interested in giving it a shot?  Looks like the new api should be more amenable.
